src/DynamoPackages/PackageLoader.cs

```diff
@@ -387,13 +388,19 @@ public void LoadCustomNodesAndPackages(IEnumerable<string> newPaths, IPreference
                 customNodeManager.AddUninitializedCustomNodesInPath(dir, false, false);
             }
 
-            foreach (var path in newPaths)
+            foreach (var path in packagePaths)
             {
-                var packageDirectory = pathManager.PackagesDirectories.FirstOrDefault(x => x.StartsWith(path));
-                if (packageDirectory != null)
+                var packageDirectory = path;
+                if (checkPathInPathManager)
```

Comment:
I see why you don't want this check in your case, but on the other hand, I think this additional flag looks confusing, and hard to understand what its purpose is.

Reply:
It is better to expose it and document it (as you noted) than perform associated code silently